python/tank/platform/engine.py

```diff
@@ -10,9 +10,10 @@
 
 """
 Defines the base class for all Tank Engines.
-
 """
 
+from __future__ import with_statement
```

Comment:
Because of the exception-safe depth tracking going on in this branch, it's easier to review it using this [link](https://github.com/shotgunsoftware/tk-core/pull/280/files?w=1) to view the changes.